### ISSUE TITLE
Spring as dependency injection framework

### ISSUE THREAD

**Issue by chrsoo** [englishtown/vertx-jersey#46]
Hi,

I am trying to figure out how to best use vertx-jersey with Spring as the dependency injection framework. Any ideas how to accomplish this without too much of a hassle?

Perhaps the easiest way is to leverage the existing HK2 spring-bridge and use HK2?

Note that I only want to use Spring for basic dependency injection to port an existing JAXRS application. The idea is not to use Spring MVC etc.

**Comment by adrianluisgonzalez** [comment]
Yeah, I think the spring-bridge is the way to go.  You can take a look at the guice-bridge for an example.

https://github.com/englishtown/vertx-jersey/blob/develop/vertx-jersey/src/main/java/com/englishtown/vertx/guice/GuiceJerseyServer.java#L39

If you want to submit a PR with Spring bridge support that would be great.

**Comment by jgleoj23** [comment]
I was able to integrate with spring dependency injection. First I copied and pasted the AutowiredInjectResolver class from [jersey's GitHub](https://github.com/jersey/jersey/blob/2.5.1/ext/spring3/src/main/java/org/glassfish/jersey/server/spring/AutowiredInjectResolver.java) into my project. Finally I just added an instance of the following Binder to the service locator:
`
import org.glassfish.hk2.api.InjectionResolver;
import org.glassfish.hk2.api.TypeLiteral;
import org.glassfish.hk2.utilities.binding.AbstractBinder;
import org.springframework.context.ApplicationContext;
import org.springframework.stereotype.Component;

/**
 * @author Joseph Gardi
 */
public class AutowiredBinder extends AbstractBinder {
    private final ApplicationContext appContext;
    public AutowiredBinder(ApplicationContext appContext) {
        this.appContext = appContext;
    }

    @Override
    protected void configure() {
        bind(new AutowiredInjectResolver(appContext))
                .to(new TypeLiteral<InjectionResolver<Component>>() {});
    }
}
`

I think that adding these classes to vert-jersey would make these integrating with spring a lot easier but it is probably not worth introducing a dependency on Spring. However, we could also put these classes in a separate library or just add an example.
